### PR TITLE
SQL + Hibernate Guide don't depend on Micronaut Data Hibernate

### LINKED ISSUE #1356: Correction in the source guide for including micronaut-data-processor
Hi Team

This ticket is to address the issue for correcting sample example to include micronaut-data-processor. 

Suggested sample example
`
              <path>
                <groupId>io.micronaut.data</groupId>
                <artifactId>micronaut-data-processor</artifactId>
                <version>4.1.3</version>
              </path>
`
Source:  https://guides.micronaut.io/latest/micronaut-jpa-hibernate-maven-java.html

Guide: https://github.com/micronaut-projects/micronaut-guides/blob/master/guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc

I have also added the version to avoid below issue. I will open PR to address the issue.

`Resolution of annotationProcessorPath dependencies failed: version can neither be null, empty nor blank`

Screenshot 2023-10-05 at 22 59 50

### PR DESCRIPTION
**Pull request by sdelamo**
See: https://github.com/micronaut-projects/micronaut-guides/issues/1356

### CODE REVIEW

**Review by timyates — APPROVED** [review]
I sometimes wonder if we should just move micronaut-sql into micronaut-data as you can't seem to use one without the other 🤔

**Comment by sdelamo** [comment]
> I sometimes wonder if we should just move micronaut-sql into micronaut-data as you can't seem to use one without the other 🤔

Yes, there is a weird cycle 

I reported an issue: 

https://github.com/micronaut-projects/micronaut-sql/issues/1157

File: guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc [hunk 1]
```diff
@@ -8,38 +8,40 @@ common:completesolution.adoc[]
...
 :dependencies:
 
 dependency:micronaut-hibernate-jpa[groupId=io.micronaut.sql,callout=1]
-dependency:micronaut-jdbc-hikari[groupId=io.micronaut.sql,callout=2]
-dependency:h2[groupId=com.h2database,scope=runtimeOnly,callout=3]
-dependency:jakarta.persistence-api[groupId=jakarta.persistence,version=2.2.3,callout=4]
+dependency:micronaut-data-tx-hibernate[groupId=io.micronaut.data,callout=2]
+dependency:micronaut-jdbc-hikari[groupId=io.micronaut.sql,callout=3]
+dependency:h2[groupId=com.h2database,scope=runtimeOnly,callout=4]
+dependency:micronaut-data-model[groupId=io.micronaut.data]
```
Comment (sdelamo) [inline guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc]:
```suggestion
```

File: guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc [hunk 2]
```diff
@@ -8,38 +8,40 @@ common:completesolution.adoc[]
...
 :dependencies:
 
 <1> Configures Hibernate/JPA EntityManagerFactory beans.
-<2> Configures SQL DataSource instances using Hikari Connection Pool.
-<3> Add dependency to in-memory H2 Database.
-<4> Add dependency to Jakarta Persistence API.
+<2> Adds Micronaut Data Transaction Hibernate dependency.
+<3> Configures SQL DataSource instances using Hikari Connection Pool.
+<4> Add dependency to in-memory H2 Database.
+<5> Add dependency to Jakarta Persistence API.
```
Comment (sdelamo) [inline guides/micronaut-jpa-hibernate/micronaut-jpa-hibernate.adoc]:
```suggestion
```

**Comment by timyates** [comment]
Something weird is going on here...

Adding

```
    annotationProcessor("io.micronaut.data:micronaut-data-document-processor")
```

fixes this...

I suspect https://github.com/micronaut-projects/micronaut-starter/pull/2173 may be wrong 🤔

**Comment by sdelamo** [comment]
Closed via https://github.com/micronaut-projects/micronaut-guides/pull/1421 and https://github.com/micronaut-projects/micronaut-guides/pull/1422

**Pull request closed without merging**